client-spark/spark2/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java

```diff
@@ -133,6 +163,15 @@ private BlockManagerId createDummyBlockManagerId(String executorId, long taskAtt
 
   @Override
   public void write(Iterator<Product2<K, V>> records) {
+    try {
+      doWrite(records);
+    } catch (Exception e) {
+      taskFailureCallback.apply(taskId);
+      throw e;
+    }
+  }
+
+  private void doWrite(Iterator<Product2<K,V>> records) {
```

Comment:
`doWrite` -> `writeImpl`?
In Reader, we have `ReaderImpl`. Do we keep consistent with it?